hs_core/management/commands/fix_broken_resources.py

```diff
@@ -74,3 +77,46 @@ def handle(self, *args, **options):
             resource_modified(r, r.creator)
             # Do a redundant check to ensure that cleaning worked.
             r.check_irods_files(echo_errors=True, log_errors=False, return_errors=False)
+
+        print("REPAIR BROKEN NETCDF RESOURCES")
+
+        for r in BaseResource.objects.filter(resource_type='NetcdfResource'):
+            if r.is_federated and not settings.REMOTE_USE_IRODS:
+                print("INFO: skipping repair of federated resource {} in unfederated mode"
+                      .format(r.short_id))
+            else:
+
+                istorage = r.get_irods_storage()
+                for f in r.files.all():
+                    try:
+                        if f.short_path.endswith('.txt') and not istorage.exists(f.storage_path):
```

Comment:
The `storage_path` includes the federation header. 

The `short_path` is what remains after you strip the federation header, the guid, and `'data/contents'`. In this case, this is the file name. If you look at the code, `short_path` goes to a bit more trouble to assure that the path is correct than `os.path.split` would...

The only reason I need both of them is that `istorage.exists` requires the `storage_path`, while the algorithm uses the `short_path` and `set_short_path`. 

However, the overhead for calling `storage_path` or `short_path` twice is fairly small.  And this routine is going to be called **once** when we roll out 1.10.0, and then **deleted** from the repository. 

I doubt that the delay from calling it twice is going to be noticed by Michael....

So, this is throwaway code, and I'd rather not optimize it further.